tracer/test/Datadog.Trace.Security.IntegrationTests/ApiSecurity/AspNetFxWebApiApiSecurity.cs

```diff
@@ -0,0 +1,91 @@
+// </copyright>
+
+#if NETFRAMEWORK
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Datadog.Trace.AppSec.Rcm.Models.Asm;
+using Datadog.Trace.AppSec.Rcm.Models.AsmData;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.RemoteConfigurationManagement;
+using Datadog.Trace.TestHelpers;
+using Xunit;
+using Xunit.Abstractions;
+
+#pragma warning disable SA1402 // File may only contain a single class
+#pragma warning disable SA1649 // File name must match first type name
+
+namespace Datadog.Trace.Security.IntegrationTests.ApiSecurity;
+
+[Collection("IisTests")]
+public class AspNetFxWebApiApiSecurityEnabled : AspNetFxWebApiApiSecurity
+{
+    public AspNetFxWebApiApiSecurityEnabled(IisFixture fixture, ITestOutputHelper output)
+        : base(fixture, output, enableApiSecurity: true)
+    {
+    }
+}
+
+[Collection("IisTests")]
+public class AspNetFxWebApiApiSecurityDisabled : AspNetFxWebApiApiSecurity
+{
+    public AspNetFxWebApiApiSecurityDisabled(IisFixture iisFixture, ITestOutputHelper output)
+        : base(iisFixture, output, enableApiSecurity: false)
+    {
+    }
+}
+
+public abstract class AspNetFxWebApiApiSecurity : AspNetBase, IClassFixture<IisFixture>
+{
+    private readonly IisFixture _fixture;
+    private readonly string _testName;
+
+    internal AspNetFxWebApiApiSecurity(IisFixture fixture, ITestOutputHelper output, bool enableApiSecurity)
+        : base("WebApi", output, "/home/shutdown", @"test\test-applications\security\aspnet")
+    {
+        SetSecurity(true);
+        if (enableApiSecurity)
+        {
+            EnvironmentHelper.CustomEnvironmentVariables.Add(ConfigurationKeys.AppSec.ApiExperimentalSecurityEnabled, "true");
+            EnvironmentHelper.CustomEnvironmentVariables.Add(ConfigurationKeys.AppSec.ApiSecurityRequestSampleRate, "1");
+        }
+
+        _fixture = fixture;
+        AddCookies(new Dictionary<string, string> { { "cookie-key", "cookie-value" } });
+        _fixture.TryStartIis(this, IisAppType.AspNetIntegrated);
+        _testName = "Security." + nameof(AspNetFxWebApiApiSecurity)
+                                + ".enableApiSecurity=" + enableApiSecurity;
+        SetHttpPort(fixture.HttpPort);
+    }
+
+    [SkippableTheory]
+    [Trait("RunOnWindows", "True")]
+    [Trait("Category", "EndToEnd")]
+    [Trait("LoadFromGAC", "True")]
+    [InlineData("/api/home/api-security/12", """{"Dog1":"23", "Dog2":"test", "Dog3": 2.5, "Dog4": 1.6}""", HttpStatusCode.OK, false)]
+    [InlineData("/api/home/api-security/12", """{"Dog1":"23", "Dog2":"dev/zero", "Dog3": 2.5, "Dog4": 1.6}""", HttpStatusCode.Forbidden, true)]
+    [InlineData("/api/home/empty-model", """{"Dog1":"23", "Dog2":"test", "Dog3": 1.5, "Dog4": 1.6}""", HttpStatusCode.OK, false)]
+    public async Task TestApiSecurityScan(string url, string body, HttpStatusCode expectedStatusCode, bool containsAttack)
```

Comment:
As a suggestion, perhaps you could add a string parameter to each test case which is only used as a name/description of the test case that could be used for the snapshot filename

Reply:
good idea, changed in [a9e27f0](https://github.com/DataDog/dd-trace-dotnet/pull/4942/commits/a9e27f0d5de510faae249110fd056c923aff97ca)

Reply:
You can also use the Verify method `.UseFileName()` which will let you encode a custom name for the file so you can get rid of the `body=` part

Reply:
oh wow I wasnt aware of it, really cool to know! I changed all file names removed body and url 🥳 I ended up using `settings.UseTextForParameters` to keep the file name's frame and change the parameters in [d84a3b4](https://github.com/DataDog/dd-trace-dotnet/pull/4942/commits/d84a3b41c49c93595ab99306479eabd354b491c1)